src/helpers/testing/mockLogger.ts

```diff
@@ -0,0 +1,51 @@
+/**
```

Comment:
Jest has [this neat little thing](https://jestjs.io/docs/manual-mocks#mocking-user-modules) where you can save a mock version of a module in a subfolder called `__mocks__`. For example, to mock `foo.ts`, you'd create `__mocks__/foo.ts`. ESLint is already configured to treat these mocks as Jest components, and TypeScript is already configured to omit these mocks from the build output like other test files.

To consume the mock in a test, each test needs only run `jest.mock("path/to/foo")`, and Jest automatically replaces `foo` with the mocked version in the adjacent `__mocks__` directory.

Reply:
That is neat. I've been messing around with it, and it looks like if I use a `__mock__` file, I'll have to use unbound methods in test files unless I rewrite the logger system. The current implementation is rather inflexible with exports.

I've converted this PR into a draft in the meantime, because it might take me a while.

Reply:
Actually, I'm convinced this is impossible without using unbounded methods somewhere.

Reply:
Need to hop on a call to talk it over?

Reply:
Never mind, I figured out a way. It's just a little hacked. I'll upload soon and we'll see what you think.